### ISSUE TITLE
Correct the distortion at all wavelengths

### ISSUE THREAD

**Issue by StevenTomczyk** [NCAR/ucomp-pipeline#185]
Description: We calibrate the distortion introduced by the UCoMP optical system (from the occulter to the cameras) by measuring the deviations in an image of a distortion mask (or grid) with equally spaced dots. The grid is placed at the location of the occulter and is imaged by both beams, so this also allows us to register the two beams. We assume that any subsequent small drifts in the x-y location of the cameras can be corrected with a small offset. 

Issue: Currently, we use distortion coefficients measured at 1074 nm to correct the distortion at all the wavelengths. This is because it is difficult to measure at other wavelengths.

Questions: Is the distortion a function of wavelength? We could check this by comparing distortion coefficients at different wavelengths? Can the distortion at other wavelengths be computed by scaling from the 1074 coefficients or some other simple means?

**Comment by jburkepile** [comment]
I took a quick look at the eccentricity of the occulters for different UCoMP wavelengths over the mission (these plots are produced by Mike). 
The plots show that 1074/1079 have the lowest eccentricity over the mission - as expected. The eccentricity is larger for the 637 nm line (also not surprising). The 789 nm line had larger eccentricity than the 637 nm line ... which was surprising. 
Also, all wavelengths show larger eccentricities in TCAM, especially starting in summer 2022. Plots are below.

![20221025 ucomp 1074 mission eccentricity](https://github.com/NCAR/ucomp-pipeline/assets/18175973/f3a1cd82-e907-4cd5-9b0b-cad309657f0e)
![20221025 ucomp 1079 mission eccentricity](https://github.com/NCAR/ucomp-pipeline/assets/18175973/db1a89b9-b86d-42d6-9777-c4ad60785a60)
![20221025 ucomp 789 mission eccentricity](https://github.com/NCAR/ucomp-pipeline/assets/18175973/18e46a96-100c-44ed-a1f8-ace53f034396)
![20221025 ucomp 637 mission eccentricity](https://github.com/NCAR/ucomp-pipeline/assets/18175973/d9db80bb-467e-4366-ba6c-44c3da24e1bf)

**Comment by detoma** [comment]
The difference between 789 and 637 is not very large and mostly in the second half of 2022 in RCAM, more striking is the difference between TCAM and RCAM.

It would be useful to have median values over a 2-3 months period to compare.

**Comment by mgalloy** [comment]
For our test date, 20220901, the RCAM and TCAM radius differ by almost a 3/4 pixel in size by the end of the day (filename, RCAM radius, TCAM radius, RCAM radius - TCAM radius):

```
20220901.182014.ucomp.1074.l1.p3.fts   333.13   333.26   -0.13
20220901.182246.ucomp.1074.l1.p3.fts   333.16   333.28   -0.12
20220901.185824.ucomp.1074.l1.p3.fts   333.19   333.32   -0.13
20220901.190056.ucomp.1074.l1.p3.fts   333.17   333.34   -0.17
20220901.193635.ucomp.1074.l1.p3.fts   333.24   333.41   -0.17
20220901.193908.ucomp.1074.l1.p3.fts   333.49   333.42    0.07
20220901.200958.ucomp.1074.l1.p3.fts   332.72   332.87   -0.15
20220901.201032.ucomp.1074.l1.p3.fts   332.81   332.97   -0.16
20220901.201106.ucomp.1074.l1.p3.fts   332.90   333.06   -0.16
20220901.201139.ucomp.1074.l1.p3.fts   332.96   333.07   -0.11
20220901.201213.ucomp.1074.l1.p3.fts   333.04   333.19   -0.15
20220901.201246.ucomp.1074.l1.p3.fts   333.04   333.19   -0.15
20220901.201320.ucomp.1074.l1.p3.fts   333.06   333.24   -0.18
20220901.201354.ucomp.1074.l1.p3.fts   333.06   333.24   -0.18
20220901.201427.ucomp.1074.l1.p3.fts   333.07   333.22   -0.15
20220901.201501.ucomp.1074.l1.p3.fts   333.13   333.29   -0.16
20220901.201534.ucomp.1074.l1.p3.fts   333.15   333.33   -0.18
20220901.201608.ucomp.1074.l1.p3.fts   333.18   333.29   -0.11
20220901.201642.ucomp.1074.l1.p3.fts   333.21   333.37   -0.16
20220901.201715.ucomp.1074.l1.p3.fts   333.26   333.42   -0.16
20220901.201749.ucomp.1074.l1.p3.fts   333.24   333.43   -0.19
20220901.201822.ucomp.1074.l1.p3.fts   333.27   333.34   -0.07
20220901.201856.ucomp.1074.l1.p3.fts   333.25   333.41   -0.16
20220901.201930.ucomp.1074.l1.p3.fts   333.26   333.40   -0.14
20220901.202003.ucomp.1074.l1.p3.fts   333.24   333.40   -0.16
20220901.202037.ucomp.1074.l1.p3.fts   333.27   333.45   -0.18
20220901.202110.ucomp.1074.l1.p3.fts   333.26   333.40   -0.14
20220901.202144.ucomp.1074.l1.p3.fts   333.34   333.42   -0.08
20220901.202218.ucomp.1074.l1.p3.fts   333.27   333.48   -0.21
20220901.202251.ucomp.1074.l1.p3.fts   333.24   333.46   -0.22
20220901.202325.ucomp.1074.l1.p3.fts   333.31   333.47   -0.16
20220901.202358.ucomp.1074.l1.p3.fts   333.31   333.45   -0.14
20220901.202432.ucomp.1074.l1.p3.fts   333.29   333.52   -0.23
20220901.202505.ucomp.1074.l1.p3.fts   333.26   333.42   -0.16
20220901.202539.ucomp.1074.l1.p3.fts   333.26   333.40   -0.14
20220901.202613.ucomp.1074.l1.p3.fts   333.30   333.41   -0.11
20220901.202646.ucomp.1074.l1.p3.fts   333.29   333.50   -0.21
20220901.202720.ucomp.1074.l1.p3.fts   333.29   333.48   -0.19
20220901.202753.ucomp.1074.l1.p3.fts   333.24   333.40   -0.16
20220901.202827.ucomp.1074.l1.p3.fts   333.29   333.41   -0.12
20220901.202901.ucomp.1074.l1.p3.fts   333.28   333.43   -0.15
20220901.202934.ucomp.1074.l1.p3.fts   333.31   333.48   -0.17
20220901.203008.ucomp.1074.l1.p3.fts   333.31   333.47   -0.16
20220901.203041.ucomp.1074.l1.p3.fts   333.29   333.44   -0.15
20220901.203115.ucomp.1074.l1.p3.fts   333.29   333.49   -0.20
20220901.203149.ucomp.1074.l1.p3.fts   333.25   333.37   -0.12
20220901.203222.ucomp.1074.l1.p3.fts   333.36   333.41   -0.05
20220901.203256.ucomp.1074.l1.p3.fts   333.28   333.55   -0.27
20220901.203329.ucomp.1074.l1.p3.fts   333.27   333.51   -0.24
20220901.203403.ucomp.1074.l1.p3.fts   333.32   333.51   -0.19
20220901.203436.ucomp.1074.l1.p3.fts   333.39   333.50   -0.11
20220901.203510.ucomp.1074.l1.p3.fts   333.27   333.49   -0.22
20220901.203544.ucomp.1074.l1.p3.fts   333.29   333.49   -0.20
20220901.203617.ucomp.1074.l1.p3.fts   333.25   333.46   -0.21
20220901.203651.ucomp.1074.l1.p3.fts   333.35   333.49   -0.14
20220901.203724.ucomp.1074.l1.p3.fts   333.21   333.41   -0.20
20220901.203758.ucomp.1074.l1.p3.fts   333.29   333.45   -0.16
20220901.203832.ucomp.1074.l1.p3.fts   333.28   333.47   -0.19
20220901.203905.ucomp.1074.l1.p3.fts   333.38   333.52   -0.14
20220901.203939.ucomp.1074.l1.p3.fts   333.39   333.53   -0.14
20220901.204012.ucomp.1074.l1.p3.fts   333.32   333.57   -0.25
20220901.204046.ucomp.1074.l1.p3.fts   333.31   333.50   -0.19
20220901.204120.ucomp.1074.l1.p3.fts   333.28   333.56   -0.28
20220901.204153.ucomp.1074.l1.p3.fts   333.22   333.42   -0.20
20220901.204227.ucomp.1074.l1.p3.fts   333.25   333.45   -0.20
20220901.204300.ucomp.1074.l1.p3.fts   333.33   333.50   -0.17
20220901.204334.ucomp.1074.l1.p3.fts   333.26   333.45   -0.19
20220901.204408.ucomp.1074.l1.p3.fts   333.22   333.44   -0.22
20220901.204441.ucomp.1074.l1.p3.fts   333.23   333.42   -0.19
20220901.204515.ucomp.1074.l1.p3.fts   333.25   333.52   -0.27
20220901.204548.ucomp.1074.l1.p3.fts   333.26   333.47   -0.21
20220901.204622.ucomp.1074.l1.p3.fts   333.30   333.47   -0.17
20220901.204656.ucomp.1074.l1.p3.fts   333.34   333.49   -0.15
20220901.204729.ucomp.1074.l1.p3.fts   333.33   333.52   -0.19
20220901.204803.ucomp.1074.l1.p3.fts   333.37   333.46   -0.09
20220901.204836.ucomp.1074.l1.p3.fts   333.33   333.54   -0.21
20220901.204910.ucomp.1074.l1.p3.fts   333.37   333.49   -0.12
20220901.204944.ucomp.1074.l1.p3.fts   333.24   333.49   -0.25
20220901.205017.ucomp.1074.l1.p3.fts   333.29   333.53   -0.24
20220901.205051.ucomp.1074.l1.p3.fts   333.24   333.41   -0.17
20220901.205124.ucomp.1074.l1.p3.fts   333.37   333.55   -0.18
20220901.205158.ucomp.1074.l1.p3.fts   333.36   333.52   -0.16
20220901.205232.ucomp.1074.l1.p3.fts   333.36   333.54   -0.18
20220901.205305.ucomp.1074.l1.p3.fts   333.28   333.48   -0.20
20220901.205339.ucomp.1074.l1.p3.fts   333.27   333.49   -0.22
20220901.205412.ucomp.1074.l1.p3.fts   333.22   333.45   -0.23
20220901.205446.ucomp.1074.l1.p3.fts   333.36   333.43   -0.07
20220901.205520.ucomp.1074.l1.p3.fts   333.30   333.56   -0.26
20220901.205553.ucomp.1074.l1.p3.fts   333.35   333.55   -0.20
20220901.205627.ucomp.1074.l1.p3.fts   333.28   333.56   -0.28
20220901.205700.ucomp.1074.l1.p3.fts   333.21   333.40   -0.19
20220901.205734.ucomp.1074.l1.p3.fts   333.26   333.41   -0.15
20220901.205808.ucomp.1074.l1.p3.fts   333.36   333.46   -0.10
20220901.205841.ucomp.1074.l1.p3.fts   333.29   333.53   -0.24
20220901.205915.ucomp.1074.l1.p3.fts   333.26   333.44   -0.18
20220901.205948.ucomp.1074.l1.p3.fts   333.27   333.41   -0.14
20220901.210022.ucomp.1074.l1.p3.fts   333.37   333.50   -0.13
20220901.210056.ucomp.1074.l1.p3.fts   333.31   333.57   -0.26
20220901.210129.ucomp.1074.l1.p3.fts   333.34   333.53   -0.19
20220901.210203.ucomp.1074.l1.p3.fts   333.28   333.52   -0.24
20220901.210236.ucomp.1074.l1.p3.fts   333.34   333.43   -0.09
20220901.210310.ucomp.1074.l1.p3.fts   333.19   333.47   -0.28
20220901.210344.ucomp.1074.l1.p3.fts   333.16   333.36   -0.20
20220901.210417.ucomp.1074.l1.p3.fts   333.20   333.39   -0.19
20220901.210451.ucomp.1074.l1.p3.fts   333.33   333.51   -0.18
20220901.210524.ucomp.1074.l1.p3.fts   333.34   333.50   -0.16
20220901.210558.ucomp.1074.l1.p3.fts   333.32   333.58   -0.26
20220901.210631.ucomp.1074.l1.p3.fts   333.29   333.54   -0.25
20220901.210705.ucomp.1074.l1.p3.fts   333.29   333.61   -0.32
20220901.210739.ucomp.1074.l1.p3.fts   333.29   333.48   -0.19
20220901.210812.ucomp.1074.l1.p3.fts   333.29   333.54   -0.25
20220901.210846.ucomp.1074.l1.p3.fts   333.28   333.56   -0.28
20220901.210919.ucomp.1074.l1.p3.fts   333.27   333.48   -0.21
20220901.210953.ucomp.1074.l1.p3.fts   333.35   333.44   -0.09
20220901.211027.ucomp.1074.l1.p3.fts   333.34   333.53   -0.19
20220901.211100.ucomp.1074.l1.p3.fts   333.37   333.59   -0.22
20220901.211134.ucomp.1074.l1.p3.fts   333.27   333.55   -0.28
20220901.211207.ucomp.1074.l1.p3.fts   333.30   333.55   -0.25
20220901.211241.ucomp.1074.l1.p3.fts   333.25   333.44   -0.19
20220901.211315.ucomp.1074.l1.p3.fts   333.25   333.42   -0.17
20220901.211348.ucomp.1074.l1.p3.fts   333.35   333.53   -0.18
20220901.211422.ucomp.1074.l1.p3.fts   333.33   333.52   -0.19
20220901.211455.ucomp.1074.l1.p3.fts   333.39   333.59   -0.20
20220901.211529.ucomp.1074.l1.p3.fts   333.25   333.49   -0.24
20220901.211602.ucomp.1074.l1.p3.fts   333.30   333.49   -0.19
20220901.211636.ucomp.1074.l1.p3.fts   333.26   333.49   -0.23
20220901.222919.ucomp.1074.l1.p3.fts   333.31   333.58   -0.27
20220901.223152.ucomp.1074.l1.p3.fts   333.32   333.62   -0.30
20220901.232814.ucomp.1074.l1.p3.fts   333.25   333.57   -0.32
20220901.233046.ucomp.1074.l1.p3.fts   333.26   333.60   -0.34
20220902.004837.ucomp.1074.l1.p3.fts   332.67   333.19   -0.52
20220902.005109.ucomp.1074.l1.p3.fts   332.72   333.21   -0.49
20220902.020509.ucomp.1074.l1.p3.fts   332.53   333.21   -0.68
20220902.020742.ucomp.1074.l1.p3.fts   332.48   333.20   -0.72
20220902.024313.ucomp.1074.l1.p3.fts   332.48   333.22   -0.74
20220902.024545.ucomp.1074.l1.p3.fts   332.55   333.24   -0.69
20220902.032124.ucomp.1074.l1.p3.fts   332.74   333.48   -0.74
20220902.032356.ucomp.1074.l1.p3.fts   332.68   333.46   -0.78
```

**Comment by bberkeyU** [comment]
The big jumps in daily scattered seem to be related to occulter size changes:
20210914: Occulter change to # 31 
20211116: Occutler changed to # 34  (also slight baffle changes around this time)
20220317: Occulter changed to # 31
20220430: Occulter changed to # 27

20220219: Shift reason unclear.

**Comment by jburkepile** [comment]
Good info Ben! Can you think of why there is no big change after the 4/30/22 occulter change? There should have been 2 more occulter changes after that. Could it be due to the fact that you purposely removed much of the occulter station hardware to reduce scattered light?

**Comment by detoma** [comment]
Mike made a special run without the distortion correction (and no hot pixels correction). The level 1 images look a lot better than the ones with the distortion correction from the wrong camera applied. We are doing significant damage to the images by applying the wrong distortion correction. This does not mean of course that we can process data without distortion. Distortion is a critical step, but the late November images were so bad looking because the used the old pipeline to process them. Examples are below.

Level 1 image with old distortion correction applied:

<img width="1280" alt="Screenshot 2024-02-15 at 9 43 12 AM" src="https://github.com/NCAR/ucomp-pipeline/assets/12946149/faf23072-504e-4f08-9b0c-c83624c3e862">


Level1 image without no distortion correction:


<img width="1276" alt="Screenshot 2024-02-15 at 9 44 23 AM" src="https://github.com/NCAR/ucomp-pipeline/assets/12946149/bd7c2055-3738-47fc-a9cf-15371b1322dc">

**Comment by bberkeyU** [comment]
Interestingly, the bad distortion seems to show some wide oops near the
pole that the no distortion seems to wash out.

I have called out the loops in the image below.

[image: image.png]

**Comment by mgalloy** [comment]
@bberkeyU Can you report above comment via the web interface? Replying via email lost your image (and also limits formatting abilities).

**Comment by bberkeyU** [comment]
![image](https://github.com/NCAR/ucomp-pipeline/assets/22605316/eaf4a919-e880-4e37-9028-ee244b5a8082)

**Comment by jburkepile** [comment]
Ben - these two images are not oriented the same way. There is a vertical flip between the two images. I asked Mike to process only 1 camera and there is no distortion applied. The vertical flip is done when the distortion is applied ... it was not done for the latest reprocessing

**Comment by bberkeyU** [comment]
Correct the post shows that the image was not flipped.
I was just pointing out that the bad distortion showed some coronal structures that the undistorted didn't. I didn't investigate this very carefully to see if this was because the structures were processing artifacts in one version and not the other. 
 Or if the wrong distortion acted like a sharpness mask and pulled out a structure washed out in the no-distortion version.

**Comment by mgalloy** [comment]
OK, I fixed the distortion step to flip the images the same way, but just not do the distortion. New images below:

![20221119 185404 ucomp 1074 l1 p3 intensity](https://github.com/NCAR/ucomp-pipeline/assets/713524/8d88133b-9e8a-43b7-bd2d-bb9c3ea43409)


Only TCAM:

![20221119 185404 ucomp 1074 l1 p3 intensity](https://github.com/NCAR/ucomp-pipeline/assets/713524/28fb15d3-e1ac-4487-abc9-334522f9b89b)

**Comment by mgalloy** [comment]
This issue requires non-1074 nm distortion data to be taken, so it can't be done until we re-open.

**Comment by jburkepile** [comment]
**Returning  to Steve's original issue for this github ticket:** 

>  'Issue: Currently, we use distortion coefficients measured at 1074 nm to correct the distortion at all the wavelengths. This is because it is difficult to measure at other wavelengths.'
> 
> Questions: Is the distortion a function of wavelength? We could check this by comparing distortion coefficients at different wavelengths? Can the distortion at other wavelengths be computed by scaling from the 1074 coefficients or some other simple means?

**Over the UCoMP mission there were 5 distortion grid tests taken:**

May 25, 2021  (doy 145)  Data taken at all wavelengths.  I believe there were significant alignment changes after this so this may be unusable.   
July 8, 2021  (doy 189)  17:09 to ?
Mar 5, 2022   (DOY 064)   ~18:58 ?  to 19:12 ? 
Nov 15, 2022 (DOY 319)  ~20:07 to 20:20ish    This appears to be a bad distortion test – grid was not well aligned and/or not in focus. Ben took other distortion test in 2024 at Giuliana’s request 
Mar 30, 3024 (DOY 090). 
 
Steve created distortion sav files from the July 8, 2021 data for these wavelengths:  789, 1074, 1079, 1083 nm
We only use the 1074 data to correct all wavelengths. 

Action Items:  
- [ ] Work with Mike to compare data at 789 nm processed with the 789.sav file and the 1074.sav file. Do the same for the 1079 nm data.   Use data between July 8, 2021 and March 4, 2022 - these dates cover the same distortion correction epoch.
- [ ] Recreate distortion correction sav files vs wavelength for the July 8, 2021  and the March 5, 2022 distortion test data.
- [ ] Dennis has said he will create the theoretical distortion correction from Zemax for the various wavelengths to assess the change in distortion with wavelength. 

I have created gif images of the distortion grid data for the July 8,2021 and Mar 5, 2022 for the 5 'synoptic' wavelengths to get a rough idea of the quality of the grid data.  The 'raw' grid data is very noisy at most/all wavelengths.  The dark-corrected distortion data may be sufficient to get a correction. Data are noisier at shorter wavelengths.
Gif images of the dark-corrected distortion grid for the 5 synoptic wavelengths for each camera are shown below. NOTE: intensity scaling may not be the best scaling to optimize contrast. 

![Image](https://github.com/user-attachments/assets/8613c23b-0e0a-4af0-8265-ec8198b6ec9e)
![Image](https://github.com/user-attachments/assets/00ffaf77-1b4b-4c18-9749-026500e7d202)

![Image](https://github.com/user-attachments/assets/70bc2ac7-eda4-4207-827e-d2ae50e0f66a)
![Image](https://github.com/user-attachments/assets/5552645e-d5ef-4952-b737-a45b33a97398)

![Image](https://github.com/user-attachments/assets/954e4188-cc79-46e7-8cd2-cd6eb8544b20)
![Image](https://github.com/user-attachments/assets/f9eaa034-0af4-481a-91b5-7f8316c74865)

![Image](https://github.com/user-attachments/assets/2d407705-aa75-4aeb-875a-b0efa7de995a)
![Image](https://github.com/user-attachments/assets/32bdcefb-9e29-4671-b53a-5cefdf8c0e6f)

![Image](https://github.com/user-attachments/assets/d40a643f-ba57-4f7c-85f8-9f3035eba6fb)
![Image](https://github.com/user-attachments/assets/2288d460-b401-4c20-b7f2-6802e01c6b16)


![Image](https://github.com/user-attachments/assets/c53a8373-1fcd-4993-a2c2-cb03cddea3d1)
![Image](https://github.com/user-attachments/assets/ea31f82a-fe8b-42d9-9efe-a568f006ec9f)

![Image](https://github.com/user-attachments/assets/dabc67c9-0509-4e12-8ca2-ce1c92b2dc2f)
![Image](https://github.com/user-attachments/assets/69392a55-9199-46ab-92bd-57cb04583eef)

![Image](https://github.com/user-attachments/assets/7329e7c0-948a-4062-a43c-bc15b3e3e988)
![Image](https://github.com/user-attachments/assets/807d154a-c47d-480a-97a5-65c04ae2944d)

![Image](https://github.com/user-attachments/assets/a80533b0-e6c1-4ad3-a572-5d34dc8959af)
![Image](https://github.com/user-attachments/assets/ff414278-fc1b-4b28-b5ec-56b59422c3cc)

![Image](https://github.com/user-attachments/assets/79c6ce67-75a6-486e-ad07-4dcfebb6b4d2)
![Image](https://github.com/user-attachments/assets/53df40b1-c532-4d6d-8cce-a4da586192a9)

**Comment by detoma** [comment]
The distortion correction is different at each wavelength, the question is by how much.  We need to assess this.

- Dennis will make a run to assess how the theoretical distortion varies with wavelength
- We need to process the grids we have at the different wavelengths. After dark correction all grids look pretty good, it seem we should have enough S/N to process all wavelengths

**Comment by detoma** [comment]
Other action items about the distortion are:

- Improve the distortion correction algorithm (this could be a good project for the new data scientist)
- Understand why there is a period in 2022 March-April when the two camera images differ by 2 pixels after distortion. Ben pointed out this seems to correspond to an occulter change, but it is not clear how the occulter change causes the two camera images to be so different. Was the occulter inserted incorrectly? 
a) At the very minimum we need to verify if the difference between the two radii is a real difference in magnification and not a failure of our centering routine. If real, we need to adjust the magnification so the camera images are properly aligned to each other.
b) If inserting a new occulter causes a large change in the distortion correction, as indicated by Ben, then a new distortion should be taken every time a new occult is inserted. This is undesirable because we plan to swap occulter more often going forward.

**Comment by bberkeyU** [comment]
Looking at the UCoMP change log, the big shift in March/April 2022 was related to a physical focus shift on TCAM around that period.

On March 17, One of the nuts that held the cameralink (data cable) into TCAM was found to have failed, causing the cable to be poorly seated.  This led to intermittent dropped frames and some of the glitched frames.  Initial attempts to fix this involved pressing the cable into the back of the camera hard (trying to get better seating). 

On April 1st the camera was removed from the instrument to to install a replacement nut. 

On April 29th, the camera, based on the results of a focus test TCAM, was driven forward by about 0.1mm. The assumption is that during the reinstallation of the camera on April 1st, the camera baffle got pinched or something, causing the camera to be pushed away from the sun by that much.  With the hindsight of a couple of years, it may just be that the stage got bumped by 0.1 mm during camera installation. 


This period of March 17-April 29th captures the differential radii pretty well.

**Comment by bberkeyU** [comment]
I created coefficients from distortion data taken on March 3, 2022, for all the wavelengths (except 530).   These files are in the ucomp-pipeline resource directory.   With the naming scheme dist_coeff_20220305_{wave}_degree4.sav.  For this run, the alignment grid was allowed to rotate to the average of the two cameras (for the 2024 distortion runs, we had no rotation).  

These distortion files were produced with a modified version of Steves's code, representing the best effort in the dot-picking parameters.  I also added the ability to force all the wave regions to have the same grid spacing; in this case, I picked the 1074.7 grid spacing.


Can we run the pipeline on a date near March 03, 2022, and look at how the rcam/tcam radii vary across wave regions? Maybe 20220306 and 20220304.

**Comment by bberkeyU** [comment]
@mgalloy  New 3-degree coefficients uploaded to GitHub/master for the native grid spacing and the 1074 grid spacing of 15.2534 pixels.

**Comment by mgalloy** [comment]
There is a run for 20220303 called `.dist-d3-s14` on the normal data server. This uses the degree 3 spacing 14.xxxx distortions.

**Comment by mgalloy** [comment]
New runs with distortion on the normal data server:

- `dist-d3-15.2534`
- `dist-d3-native` (currently running at 7:30 pm MT, but there are L1 files of every wave region already)

**Comment by bberkeyU** [comment]
Pipeline reported camera occulter RAIDUS from the fits headers: for the old distortion, new distortion with native grid spacing, and forced 1074 grid spacing.

Radius mean : stddev values for 2022-03-03 
Filter 	 1074 space 	 native space 	 old process 
637.0 	340.14 : 0.21 	332.63 : 0.25 	 334.13 : 0.16 
706.0 	340.65 : 0.18 	334.52 : 0.11 	 335.54 : 0.20 
789.0 	340.45 : 0.13 	335.62 : 0.12 	 337.06 : 0.13 
1074.0 	339.90 : 0.11 	339.90 : 0.11 	 341.76 : 0.09 
1079.0 	339.91 : 0.20 	339.91 : 0.20 	 341.83 : 0.13 

![old distortion](https://github.com/user-attachments/assets/b57ce935-662c-4c34-b08e-8f6d16e731ab)

![new distortion with native grid spacing](https://github.com/user-attachments/assets/37323371-c842-4dea-a6c9-495abcde4cb3)

![new distortion with fixed 1074 spacing](https://github.com/user-attachments/assets/c96b1c0e-48b2-40fa-b7e1-778ccfcd6f51)

**Comment by detoma** [comment]
Ben, can you clarify what these numbers mean?

Radius mean : stddev values for 2022-03-03
Filter 1074 space native space old process
637.0 340.14 : 0.21 332.63 : 0.25 334.13 : 0.16
706.0 340.65 : 0.18 334.52 : 0.11 335.54 : 0.20
789.0 340.45 : 0.13 335.62 : 0.12 337.06 : 0.13
1074.0 339.90 : 0.11 339.90 : 0.11 341.76 : 0.09
1079.0 339.91 : 0.20 339.91 : 0.20 341.83 : 0.13

In particular:
- What are the old and new distortions? I am guessing "old" distortion is your distortion using the different grid wavelengths and mapping them into a common grid, and the "new" is your distortion using the different grid wavelengths but using the average spacing like in the original code. If so, can you please write it down.  This is hard to follow even for somebody who has been in our meetings
- I do not understand the format you used. What is the meaning of ":" For example in the top raw, does 340.14:0.21 means 340.14+/- 0.21? 
-  Did you use the average value of the radius? We need to check the values for the two cameras before the radii is averaged together.

**Comment by detoma** [comment]
When everything is mapped on the same grid, the difference in radius between wavelengths is larger than we would like to see. However, it is unclear how much of this difference is caused by the uncertainty in the distortion correction and how much is our error in finding the center/radius at the shorter wavelengths.

I think we need to compare the corrected grid images to assess the accuracy of the distortion correction.

**Comment by bberkeyU** [comment]
> * What are the old and new distortions? I am guessing "old" distortion is your distortion using the different grid wavelengths and mapping them into a common grid, and the "new" is your distortion using the different grid wavelengths but using the average spacing like in the original code. If so, can you please write it down.  This is hard to follow even for somebody who has been in our meetings

Plots showed the values from the headers of process, process.dist-d3-15.2534, and process.dist-d3-native.  Where old distortion referees to the data from the process directory.  The new native spacing runs the distortion code mapping to a grid spacing based on the distance between points in each wave region.  The new 1074 spacing computes a distortion map to 1074.7 grid spacing of 15.2534pixels.

> * I do not understand the format you used. What is the meaning of ":" For example in the top raw, does 340.14:0.21 means 340.14+/- 0.21?
In this case the mean of the points was 340.14 and the standard deviation was 0.21

> * Did you use the average value of the radius? We need to check the values for the two cameras before the radii is averaged together.
This table and plots use the RADIUS value from the header which I think is the average of RADIUS0 and RADIUS1

**Comment by bberkeyU** [comment]
> When everything is mapped on the same grid, the difference in radius between wavelengths is larger than we would like to see. However, it is unclear how much of this difference is caused by the uncertainty in the distortion correction and how much is our error in finding the center/radius at the shorter wavelengths.
> 
> I think we need to compare the corrected grid images to assess the accuracy of the distortion correction.

I was also hoping to see a better agreement across the wave regions when the distortions were mapped to the same grid, and I also don't know where the error is coming from.   However, I dont know how (in a mathematically rigorous sense) you would like to try and assess the accuracy of the distortion correction.
